### ISSUE TITLE
error: Invoice number must be unique

### ISSUE THREAD

**Issue by izeblood** [opensourcepos/opensourcepos#2578]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [] Reporting an issue of an unmodified OSPOS installation
- [] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 
- OSPOS git commit hash is: 
- PHP version is: 
- MySQL or MariaDB version is: 
- OS and version is: 
- WebServer is: 
- Selected language is: 
- (If applicable) Docker installation: 
- (If applicable) Installation package for the LAMP/LEMP stack is: 

#Issue #Bug  #Question 
@jekkos @odiea @daN4cat @SteveIreland 
I install ospos3.3.3 0b9a76 , I changed the Register mode: to Invoice and when I click Invoice button, this message appear "Invoice number must be unique" kindly give me some enlightenment please

![error](https://user-images.githubusercontent.com/32444665/66617527-0a681e80-ec08-11e9-84b7-8ac9a43d856d.jpg)

**Comment by izeblood** [comment]
there's a bug on the invoice

**Comment by SteveIreland** [comment]
Refer to https://github.com/opensourcepos/opensourcepos/wiki/Sales in the section for Sales Invoice Format.

How the invoice number is assigned depends on how you have it configured and you can certainly configure it to create duplicate invoice numbers if you're not careful.

**Comment by izeblood** [comment]
yes I know this...kindly try it by yourself the version 3.3.0 0b9a76 .. Try it then you found out there's something wrong on it... because the previous version is OK.

**Comment by SteveIreland** [comment]
Did you close this issue because you figured out what the issue was?

I ask questions because I'm always testing with the latest version and haven't stumbled across this issue.

So I'll need a bit more info in order to avoid wasting my time.  Is this problem occurring and a clean database or a migrated database?  Which of the invoice formatting codes are you using?

**Comment by mhanson-vertis** [comment]
@SteveIreland this bug still persists in Open Source Point Of Sale version 3.3.1 - c786d4.

I was able to get around it by modifying Office -> Configuration -> Invoice -> Sales Invoice Format.

Initial value was {CO}; final value was INV-%y-{CO}. It appears the application is failing to ensure a unique invoice number by default, or simply fails a check upon attempting to generate the first invoice in it's default configuration.

NB. After generating the first couple invoices I switched back to the default configuration (Sales Invoice Format = {CO}) and was able to continue generating invoices.
